lib/global_registry/base.rb

```diff
@@ -4,26 +4,46 @@
 
 module GlobalRegistry
   class Base
+    def initialize(args = {})
+      @base_url = args[:base_url]
+      @access_token = args[:access_token]
+      @xff = args[:xff]
+    end
 
     def self.find(id, params = {})
```

Comment:
The class level methods should probably all be rewritten to use the instance level methods now. E.g:

```ruby
def self.find(id, params = {})
  new.find(id, params)
end
```